File: 프로그래머스/0/181918. 배열 만들기 4/배열 만들기 4.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> stk;
    for(int i = 0; i < arr.size(); )
    {
        if(i < arr.size())
        {
            if(stk.size() == 0)
            {
                stk.push_back(arr[i++]);   
            }
            else
            {
                if(stk[stk.size() - 1] < arr[i])
                    stk.push_back(arr[i++]);
                else
                    stk.pop_back();
            }
        }
    }
    return stk;
}

File: 프로그래머스/0/181918. 배열 만들기 4/배열 만들기 4.cs
using System;
using System.Collections.Generic;
public class Solution {
    public int[] solution(int[] arr) {
        List<int> stk = new List<int>();
        for(int i = 0; i < arr.Length; )
        {
            if(i < arr.Length)
            {
                if(stk.Count == 0)
                {
                    stk.Add(arr[i++]);   
                }
                else
                {
                    if(stk[stk.Count - 1] < arr[i])
                        stk.Add(arr[i++]);
                    else
                        stk.RemoveAt(stk.Count - 1);
                }
            }
        }
        return stk.ToArray();
    }
}